plugin/secrets_access_token.go

```diff
@@ -182,3 +104,63 @@ func (tg *TokenGenerator) getAccessToken(ctx context.Context, iamAdmin *iam.Serv
 	}
 	return tkn, err
 }
+
+const deprecationWarning = `
+We have recently changed the behavior of the gcp/token/ endpoint to no longer generate leases. 
+This is due to limitations of the GCP API, as service account OAuth access tokens cannot be revoked.
+This access_token secret and associated lease were created by an older version of the GCP secrets engine
+and will be cleaned up now. Note that there is the chance that this access_token, if not already expired, 
+will still be valid up to one hour. 
+`
+
+const pathTokenHelpSyn = `Generate an OAuth2 access token under a specific role set.`
+const pathTokenHelpDesc = `
+This path will generate a new OAuth2 access token for accessing GCP APIs.
+A role set, binding IAM roles to specific GCP resources, will be specified 
+by name - for example, if this backend is mounted at "gcp",
+then "gcp/token/deploy" would generate tokens for the "deploy" role set.
+
+On the backend, each roleset is associated with a service account. 
+The token will be associated with this service account. Tokens have a 
+short-term lease (1-hour) associated with them but cannot be renewed.
+
+NOTE: As of Vault 0.11.2, we have changed the behavior of the token/ endpoint.
+Note that revocation is no longer supported and Vault will no longer create a
+lease for this type. 
+
+Please see backend documentation for more information: 
+https://www.vaultproject.io/docs/secrets/gcp/index.html
+`
+
+// EVERYTHING USING THIS SECRET TYPE IS CURRENTLY DEPRECATED.
+// We keep it to allow for clean up of access_token secrets/leases that may have be left over
+// by older versions of Vault.
+const SecretTypeAccessToken = "access_token"
+
+func secretAccessToken(b *backend) *framework.Secret {
+	return &framework.Secret{
+		Type: SecretTypeAccessToken,
+		Fields: map[string]*framework.FieldSchema{
+			"token": {
+				Type:        framework.TypeString,
+				Description: "OAuth2 token",
+			},
+		},
+		Renew:  b.secretAccessTokenRenew,
+		Revoke: b.secretAccessTokenRevoke,
+	}
+}
+
+// Renewal will still return an error, but return the warning in case as well.
+func (b *backend) secretAccessTokenRenew(ctx context.Context, req *logical.Request, d *framework.FieldData) (*logical.Response, error) {
+	resp := logical.ErrorResponse("short-term access tokens cannot be renewed - request new access token instead")
+	resp.AddWarning(deprecationWarning)
+	return resp, nil
+}
+
+// Revoke will no-op and pass but warn the user. This is mostly to clean up old leases.
+// Any associated secret (access_token) has already expired and thus doesn't need to
+// actually be revoked,  or will expire within an hour and currently can't actually be revoked anyways.
+func (b *backend) secretAccessTokenRevoke(ctx context.Context, req *logical.Request, d *framework.FieldData) (*logical.Response, error) {
+	return &logical.Response{Warnings: []string{deprecationWarning}}, nil
```

Comment:
Nit: I think we should always use the `AddWarning` API, since the field structs aren't technically part of the "API".

```go
var resp logical.Response
resp.AddWarning(deprecationWarning)
return &resp
```